lib/nn.js

```diff
@@ -9,6 +9,13 @@ function dsigmoid(y) {
   return y * (1 - y);
 }
 
+function tanh(x) {
+	return Math.tanh(x);
+}
+function dtanh(x) {
+	return 1-(tanh(x)*tanh(x));
```

Comment:
Since the output of a node has already been passed through `tanh()` I believe this would be `(1-y*y)`? (The convention I'm using to note this is `y` instead of `x`. Am I mistaken about this?